osrf_testing_tools_cpp/include/osrf_testing_tools_cpp/memory_tools/memory_tools_service.hpp

```diff
@@ -93,6 +93,11 @@ struct MemoryToolsService
   void
   print_backtrace();
 
+  /// Returns a backtrace stream for logging.
+  OSRF_TESTING_TOOLS_CPP_MEMORY_TOOLS_PUBLIC
+  std::ostringstream
+  get_backtrace();
```

Comment:
I don't love the name `get_backtrace`; it is unclear what you are getting.  How about `get_backtrace_stream`?  Also, I think this method can be `const`.